test/helpers/pullOutValues.js

```diff
@@ -0,0 +1,255 @@
+const assert = require('assert')
+
+const { Ok } = require('../../src/results')
+const { pullOutValues } = require('../../src/helpers')
+
+describe('Helpers - pullOutValue', () => {
```

Comment:
great. but where is the test for the scenario that motivated this refactoring: circular references?

Reply:
Of course, my bad. I will create that test.